src/Paket.Core/PackageResolver.fs

```diff
@@ -257,313 +283,489 @@ type UpdateMode =
+        }
+
+        let conflictState =
+            if Seq.isEmpty compatibleVersions then
+                boostConflicts currentStep.FilteredVersions currentRequirement knownConflicts conflictHistory conflictState
+            else
+                conflictState
+
+        let flags =
+            StepFlags
+                (   ready       = false
+                ,   useUnlisted = false
+                ,   hasUnlisted = false
+                ,   forceBreak  = flags.ForceBreak
+                ,   firstTrial  = flags.FirstTrial
+                )
+
+        let rec stepLoop (flags:StepFlags) (stage:Stage) =
+            match stage with
+            | Outer (conflictState) ->
+                if flags.Ready then conflictState else
+                let flags = StepFlags(flags.Ready,flags.UseUnlisted,flags.HasUnlisted,false,true)
+                stepLoop flags (Inner(conflictState))
+
+            | Inner (conflictState) ->
+                if stopLooping flags conflictState then
+                    let flags =
+                        if not flags.UseUnlisted && flags.HasUnlisted && not conflictState.Status.IsDone then
+                            StepFlags(flags.Ready,true,flags.HasUnlisted,flags.ForceBreak,flags.FirstTrial)
+                        else
+                            StepFlags(true,flags.UseUnlisted,flags.HasUnlisted,flags.ForceBreak,flags.FirstTrial)
+                    stepLoop flags (Outer(conflictState))
+                else
+                let flags = StepFlags(flags.Ready,flags.UseUnlisted,flags.HasUnlisted,flags.ForceBreak,false)
+                let (version,sources) & versionToExplore = Seq.head conflictState.VersionsToExplore
+
+                let conflictState = {
+                    conflictState with
+                        VersionsToExplore = Seq.tail conflictState.VersionsToExplore
+                }
+                let packageDetails = {
+                    GroupName          = groupName
+                    Dependency         = currentRequirement
+                    GlobalRestrictions = globalFrameworkRestrictions
+                    RootSettings       = rootSettings
+                    Version            = version
+                    Sources            = sources
+                    UpdateMode         = updateMode
+                }
+
+                let exploredPackages = conflictState.ExploredPackages
+
+                match getExploredPackage packageDetails getPackageDetailsF exploredPackages with
+                | None ->
+                    stepLoop flags (Inner(conflictState))
+                | Some exploredPackage ->
+                    let hasUnlisted = exploredPackage.Unlisted || flags.HasUnlisted
+
+                    let flags = StepFlags(flags.Ready,flags.UseUnlisted,hasUnlisted,flags.ForceBreak,flags.FirstTrial)
+                    if exploredPackage.Unlisted && not flags.UseUnlisted then
+                        tracefn "     unlisted"
```

Comment:
now i'm pretty sure the bug is here, where I last made a change :facepalm:
which I'd had as 
```fsharp
                    if exploredPackage.Unlisted && not flags.UseUnlisted then
                        tracefn "     unlisted"
                        stepLoop flags (Inner(conflictState))
                    else
```
before, but that caused an infinite loop printing `unlisted`

@forki it'd be really useful for whoever has to work on this next, if you or someone else with a deeper understanding of what's going on in this module could add some comments explaining the processes. I'd do it myself, but I don't think I have a firm enough grasp and I wouldn't want to put down something that's wrong